### DISCUSSION TITLE
Cannot export when using null coalescing

### DISCUSSION THREAD

**Discussion by jsok** [cue-lang/cue#1567] [Q&A] [answered]
I'm attempting to use [null coalescing](https://cuelang.org/docs/tutorials/tour/expressions/coalesce/) to group some values, a minimal reproduction can be found here: https://cuelang.org/play/?id=BbkhOFqxreI#cue@export@cue

It's effectively the same as this [discussion 679](https://github.com/cue-lang/cue/discussions/679) except I'm attempting to group by a value which might not be set, and use a fallback value instead.

Exporting to CUE produces the expected `grouped` value , but attempting to export to JSON/YAML fails with an incomplete value error. Is this expected?

**Comment by jlongtine** [comment, marked as answer]
@jsok It's failing because your "schema" `resource` isn't a [definition](https://cuelang.org/docs/tutorials/tour/types/defs/), so it's trying to export it as JSON/YAML (which it can't, because it has incomplete fields, namely `name: string`). 

The easiest way to fix that is to change `resource` -> `#resource`, as shown here: https://cuelang.org/play/?id=-kW9M3P9dSS#cue@export@json

That correctly exports as JSON. Hope that helps!

One thing to note: If you need `resource` to be open, because you'll want to add other fields (beyond `metadata`), you can keep it open like this:

```cue
#resource: {
	metadata: {
		name: string
		labels: [string]: string
	}
	...
}
```

**Reply by jsok** [reply]
@jlongtine thanks that does seem to fix the issue.

I'm not understanding why `resource.metadata.name: incomplete value string` is happening though, every `resource` has a `metadata.name` so how can the value be incomplete?

For a bit more context, the config I'm working with is closely modelled on https://github.com/cue-lang/cue/blob/v0.4.2/doc/tutorial/kubernetes/README.md, so this reproduction is based on issues faced when trying to export resources defined in that style. I note at least in the "quick n dirty" sections of the tutorial there isn't any **definitions** but just structs. Is that just generally bad style?
